### PR TITLE
Add the webhook image

### PR DESCRIPTION
**Pull request by manuelbuil**
Signed-off-by: Manuel Buil <mbuil@suse.com>

### CODE REVIEW

File: Dockerfile.webhook [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    make \
+    patch
+ARG TAG
+RUN git clone https://github.com/k8snetworkplumbingwg/sriov-network-operator
+WORKDIR sriov-network-operator
+RUN git fetch --all --tags --prune
+RUN git checkout remotes/origin/${TAG} -b ${TAG}
+COPY 0001-CGO_ENABLED.patch 0002-Change-the-module-URL-from-openshift-to-k8snetworkpl.patch ./
+RUN patch -p1 < 0001-CGO_ENABLED.patch
+RUN patch -p1 < 0002-Change-the-module-URL-from-openshift-to-k8snetworkpl.patch
```
Comment (mrostecki) [inline Dockerfile.webhook:18]: I don't really like the fact that we are doing all those `git` and `patch` operations as separate `RUN` instructions, which means that we are creating a layer for each of them. Every single command is creating a new layer which is taking space.

Usually the good practice I like is doing the whole git repo fetching, build, installation and then removing the unneeded source dir in one `RUN` instruction. For example:

```
RUN git clone http://github.com/foo/bar \
    && cd bar \
    && make -j$(proc) \
    && make install \
    && cd .. \
    && rm -rf bar
```

In that way, you are creating just a single filesystem layer with just a binary. Sources are used temporality only for building the binary, but are not bloating the resulting fs layer.

If we've already done something like that in other Rancher images, I think that's a tech debt to fix.

File: README.md [hunk 2]
```diff
@@ -1 +1,26 @@
...
+
+```
+Volumes:
+  tls:
+    Type:        Secret (a volume populated by a Secret)
+    SecretName:  operator-webhook-service
+    Optional:    false
+```
+
+When doing this manually, the secret must be create using:
```
Comment (mrostecki) [inline README.md:13]:
```suggestion
When doing this manually, the secret must be created using:
```

**Review by mrostecki — CHANGES_REQUESTED** [review]

File: Dockerfile.webhook [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ENV CGO_ENABLED=0
+RUN make _build-webhook
```
Comment (mrostecki) [inline Dockerfile.webhook:20]: Does the `_build-webhook` target also install the binary in `/usr/bin`? It seems like, but that's really cofusing...

File: Dockerfile.webhook (hunk 1 above)
Comment (manuelbuil) [reply Dockerfile.webhook:18]: That's a great point. Thanks! Let me fix that

Comment (manuelbuil) [reply Dockerfile.webhook:18]: Reg. sources, note that we are not releasing the image where we build stuff. We are copying the binaries that the ubi image

File: Dockerfile.webhook [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ARG TAG
+ENV CGO_ENABLED=0
+COPY 0001-CGO_ENABLED.patch 0002-Change-the-module-URL-from-openshift-to-k8snetworkpl.patch ./
+RUN git clone https://github.com/k8snetworkplumbingwg/sriov-network-operator \
+    && cd sriov-network-operator \
+    && git fetch --all --tags --prune \
+    && git checkout remotes/origin/${TAG} -b ${TAG} \
+    && patch -p1 < ../0001-CGO_ENABLED.patch \
+    && patch -p1 < ../0002-Change-the-module-URL-from-openshift-to-k8snetworkpl.patch \
+    && make _build-webhook
```
Comment (mrostecki) [inline Dockerfile.webhook:20]:
```suggestion
    && make _build-webhook \
    && cd .. \
    && rm -rf sriov-network-operator
```

**Review by mrostecki — CHANGES_REQUESTED** [review]
_comment removed, please disregard_

File: Dockerfile.webhook (hunk 3 above)
Comment (manuelbuil) [reply Dockerfile.webhook]: It installs the binary in build/_output/linux/amd64/webhook. Then we copy that to `/usr/bin/webhook` in the UBI_IMAGE

**Review by mrostecki — APPROVED** [review]
LGTM, thanks!

**Pull request merged**